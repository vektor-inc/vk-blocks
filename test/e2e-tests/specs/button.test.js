/**
 * WordPress dependencies
 */
 import {
    createNewPost,
    getEditedPostContent
} from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import { changeSiteLang } from '../helper/changeSiteLang';

describe( 'Button', () => {
    let oldLanguage;
	beforeEach( async () => {
        oldLanguage = await changeSiteLang( 'ja' );
		await createNewPost();
	} );
    it( 'VK Button Block ( single )', async () => {
        //Using the slash command
		await page.keyboard.press( 'Enter' );
        await page.keyboard.type( '/ボタン' );
		// await page.waitForXPath(
        //     `//*[contains(@class, "components-autocomplete__result") and contains(@class, "is-selected") and contains(text(), 'ボタン')]`
        // );
		await page.click( 'button.components-button[id$="vk-blocks/button"]' );
		await page.keyboard.type( 'VK ボタン' );
		// id入力フォームに一旦フォーカス
		await page.click( '.is-sidebar-opened .components-panel__body.is-opened .components-text-control__input[type="text"]' );
		// idを削除
		await page.$eval( '.is-sidebar-opened .components-panel__body.is-opened .components-text-control__input[type="text"]', element => element.value = '');
		// idを手入力
		await page.keyboard.type( 'test-button' );
        expect( await getEditedPostContent() ).toMatchSnapshot();
    } )
});