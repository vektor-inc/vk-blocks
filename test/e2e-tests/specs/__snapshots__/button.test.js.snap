// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button VK Button Block ( single ) 1`] = `
"<!-- wp:vk-blocks/button {\\"blockId\\":\\"test-button\\"} -->
<div class=\\"wp-block-vk-blocks-button vk_button vk_button-color-custom vk_button-align-left\\"><a class=\\"vk_button_link btn has-background has-vk-color-primary-background-color btn-md\\" role=\\"button\\" aria-pressed=\\"true\\" rel=\\"noopener\\"><span class=\\"vk_button_link_txt\\">VK ボタン</span></a></div>
<!-- /wp:vk-blocks/button -->"
`;